scope3_methodology/ad_tech_platform/model.py

```diff
@@ -0,0 +1,328 @@
+        self.datacenter_water_intensity_h2o_m_3_per_mwh = datacenter_water_intensity_h2o_m_3_per_mwh
+        self.server_emissions_g_per_kwh = server_emissions_g_per_kwh
+        self.servers_processing_cookie_syncs_percentage = servers_processing_cookie_syncs_percentage
+        self.cookie_sync_distribution_rate = cookie_sync_distribution_rate
+
+    def get_bid_requests_processed_per_month(self) -> float:
+        return self.bid_requests_processed_billion_per_month * 1000000000
+
+    def get_server_emissions_g_co2e_per_month(self) -> float:
+        return self.server_emissions_mt_co2e_per_month * 1000000
+
+    def get_servers_processing_bid_requests_pct(self) -> float:
+        return self.servers_processing_bid_requests_percentage / 100.0
+
+    def get_servers_processing_cookie_syncs_pct(self) -> float:
+        return self.servers_processing_cookie_syncs_percentage / 100.0
+
+    def get_cookie_syncs_processed_per_month(self) -> float | None:
+        if self.cookie_syncs_processed_billion_per_month:
+            return self.cookie_syncs_processed_billion_per_month * 1000000000
+        return None
+
+    def get_corporate_allocation_decimal_fraction(self) -> float:
+        return self.corporate_allocation_percentage / 100.0
+
+    def get_servers_processing_cookie_syncs_decimal_fraction(self) -> float:
+        return self.servers_processing_cookie_syncs_percentage / 100.0
+
+    def get_servers_processing_bid_requests_decimal_fraction(self) -> float:
+        return self.servers_processing_bid_requests_percentage / 100.0
+
+    def get_decimal_fraction_of_bid_requests_processed_from_ad_tech_platforms(self) -> float:
+        return self.percentage_of_bid_requests_processed_from_ad_tech_platforms / 100.0
+
+    def get_data_transfer_emissions_g_co2e_per_month(self) -> float | None:
+        if self.data_transfer_emissions_mt_co2e_per_month:
+            return self.data_transfer_emissions_mt_co2e_per_month * 1000000
+        return None
+
+    def get_server_allocation_decimal_fraction(self) -> float:
+        return self.server_allocation_percentage / 100.0
+
+    def comp_external_request_ratio(self, depth: int) -> float:
+        if (
+            self.ad_tech_platform_bid_requests_processed_billion_per_month
+            and self.bid_requests_processed_billion_per_month
+        ):
+            ratio = (
+                self.ad_tech_platform_bid_requests_processed_billion_per_month
+                / self.bid_requests_processed_billion_per_month
+            )
+            log_result("external request ratio", f"{ratio * 100.0:.1f}%", depth)
+            return ratio
+        return self.get_decimal_fraction_of_bid_requests_processed_from_ad_tech_platforms()
+
+    def comp_bid_request_size_gb(self) -> float:
+        return self.bid_request_size_in_bytes / 1024 / 1024 / 1024
+
+    def comp_data_transfer_emissions_g_co2e_per_bid_request(self, depth: int) -> float:
```

Comment:
Have some questions about this function. Existing code is https://github.com/scope3data/methodology/blob/main/scope3_methodology/adtech_model.py#L125

**NOTE** This old function used to return `mt` if we had "data transfer emissions mt per month" and `g` if we had to use other facts to compute this number. Moved this to always return `g`. 

Why aren't we returning this divided by `get_bid_requests_processed_per_month()` ? Right now, this doesn't seem like its actually per bid request but the total emissions from data transfer

Reply:
As far as I can tell you are right - looks like a bug in my code